packages/core-data/src/reducer.js

```diff
@@ -529,37 +534,41 @@ export function undo( state = UNDO_INITIAL_STATE, action ) {
 			};
 		}
 
-		case 'EDIT_ENTITY_RECORD':
-			// When undo is not specified,
-			// It's unclear whether we should ignore the undo entirely
-			// or consider it a transient (cached) edit.
-			if ( ! action?.meta?.undo ) {
+		case 'EDIT_ENTITY_RECORD': {
+			if ( ! action.meta.undo ) {
 				return state;
 			}
 
 			const isCachedChange = Object.keys( action.edits ).every(
 				( key ) => action.transientEdits[ key ]
 			);
 
+			const edits = Object.keys( action.edits ).map( ( key ) => {
+				return {
+					kind: action.kind,
+					name: action.name,
+					recordId: action.recordId,
+					property: key,
+					from: action.meta.undo.edits[ key ],
+					to: action.edits[ key ],
+				};
+			} );
+
 			if ( isCachedChange ) {
+				let newCache = state.cache;
+				edits.forEach(
+					( edit ) =>
+						( newCache = appendEditToStack( newCache, edit ) )
+				);
```

Comment:
Use a reducer pattern here:
```js
const newCache = edits.reduce( appendEditToStack, state.cache );
```
There are multiple opportunities to this, e.g., the `stackedEdits.forEach` loop could be turned into `stackedEdits.reduce`.

Reply:
I like less and less reduce to generate objects based on arrays personally, I find that code less readable.

Reply:
You're right that it's a great fit for this one though :)

Reply:
If you mean the popular pattern where we do:
```js
items.reduce( ( acc, [ key, value ] ) => {
  acc[ key ] = value;
  return acc;
}, {} );
```
then I totally agree with you, doing a for loop or using `Object.fromEntries` is much better.

But our case is different, `appendEditsToStack` _is_ a reducer function, and applying it with `.reduce` makes perfect sense, that's _exactly_ what `.reduce` is for.